nvflare/app_common/ccwf/cse_server_ctl.py

```diff
@@ -0,0 +1,250 @@
+from nvflare.app_common.ccwf.val_result_manager import EvalResultManager
+from nvflare.fuel.utils.validation_utils import (
+    DefaultPolicy,
+    check_positive_number,
+    check_str,
+    validate_candidate,
+    validate_candidates,
+)
+
+
+class CrossSiteEvalServerController(ServerSideController):
+    def __init__(
+        self,
+        task_name_prefix=Constant.TN_PREFIX_CROSS_SITE_EVAL,
+        start_task_timeout=Constant.START_TASK_TIMEOUT,
+        configure_task_timeout=Constant.CONFIG_TASK_TIMEOUT,
+        eval_task_timeout=30,
+        task_check_period: float = Constant.TASK_CHECK_INTERVAL,
+        job_status_check_interval: float = Constant.JOB_STATUS_CHECK_INTERVAL,
+        progress_timeout: float = Constant.WORKFLOW_PROGRESS_TIMEOUT,
+        participating_clients=None,
+        evaluators=None,
+        evaluatees=None,
+        global_model_client=None,
+        max_status_report_interval: float = Constant.PER_CLIENT_STATUS_REPORT_TIMEOUT,
+        eval_result_dir=AppConstants.CROSS_VAL_DIR,
+    ):
+        if not evaluatees:
+            evaluatees = []
+
+        if not evaluators:
+            evaluators = []
+
+        super().__init__(
+            num_rounds=1,
+            task_name_prefix=task_name_prefix,
+            start_task_timeout=start_task_timeout,
+            configure_task_timeout=configure_task_timeout,
+            task_check_period=task_check_period,
+            job_status_check_interval=job_status_check_interval,
+            participating_clients=participating_clients,
+            starting_client="",
+            starting_client_policy=DefaultPolicy.EMPTY,
+            starting_client_allow_none=True,
+            max_status_report_interval=max_status_report_interval,
+            result_clients=None,
+            result_clients_policy=DefaultPolicy.EMPTY,
+            result_clients_allow_none=True,
+            progress_timeout=progress_timeout,
+        )
+
+        check_str("eval_result_dir", eval_result_dir)
+        check_positive_number("eval_task_timeout", eval_task_timeout)
+
+        if not global_model_client:
+            global_model_client = ""
+        self.global_model_client = global_model_client
+        self.eval_task_name = make_task_name(task_name_prefix, Constant.BASENAME_EVAL)
+        self.eval_task_timeout = eval_task_timeout
+        self.eval_local = False
+        self.eval_global = False
+        self.evaluators = evaluators
+        self.evaluatees = evaluatees
+        self.eval_result_dir = eval_result_dir
+        self.global_names = {}
+        self.eval_manager = None
+        self.current_round = 0
+
+    def start_controller(self, fl_ctx: FLContext):
+        super().start_controller(fl_ctx)
+
+        self.evaluators = validate_candidates(
+            var_name="evaluators",
+            candidates=self.evaluators,
+            base=self.participating_clients,
+            default_policy=DefaultPolicy.ALL,
```

Comment:
DefaultPolicy means "the policy for handling default value (when user doesn't not explicitly configure the value)", not the "default policy when not policy is specified".

I know this could be a little confusing. Maybe we'll just rename it to PolicyForDefaultValueHandling. It's a bit long but more clear.

Reply:
Changed to use DefaultValuePolicy.